src/ameedf/jb8223/assignments/hw02/Clock.java

```diff
@@ -34,4 +34,106 @@
  * BONUS 2: the same functionality as in "BONUS 1" while using as few member variables / memory as possible
  */
 public class Clock {
+
+    private int hours;
+    private int minutes;
+    private int seconds;
+    private int milliseconds;
+
+    private final int minValueAllowed = 0;
+    private final int maxValueAllowed = 1_000_000_000;
```

Comment:
if you want to define the variable as final, then you should also define it as static